EIPS/eip-4910.md

```diff
@@ -0,0 +1,1111 @@
+---
+eip: 4910
+title: Royalty Bearing NFTs
+description: Extension of the ERC-721 standard for NFTs to correctly define, process, and pay (hierarchical) onchain royalties from NFT sales, going beyond EIP-2981.
+author: Andreas Freund (@Therecanbeonlyone1969) 
+discussions-to: https://ethereum-magicians.org/t/royalty-bearing-nfts/8453
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-03-14
+requires: 721, 165
+---
+
+## Abstract
+The proposal directly connects NFTs and royalties in a smart contract architecture extending the [ERC-721](./eip-721.md) standard, with the aim of precluding central authorities from manipulating or circumventing payments to those who are legally entitled to them.
+
+The proposal builds upon the [OpenZeppelin Smart Contract Toolbox](https://github.com/OpenZeppelin/openzeppelin-contracts) architecture, and extends it to include royalty account management (CRUD), royalty balance and payments management, simple trading capabilities -- Listing/De-Listing/Buying -- and capabilities to trace trading on exchanges. The royalty management capabilities allow for hierarchical royalty structures, referred to herein as royalty trees, to be established by logically connecting a "parent" NFT to its "children", and recursively enabling NFT "children" to have more children. 
```

Comment:
Based on the... heft of the reference implementation, and this link to OZ, I have to ask: is it likely that there will be multiple implementations of this standard? If not, would this be better as a PR into OZ itself?

Reply:
@SamWilsn there will be more than one implementation of this standard. I think it will eventually make it into OZ via a PR though.